mayastor/tests/lock_lba_range.rs

```diff
@@ -0,0 +1,269 @@
+    Arc::new(Mutex::new(nexus.get_channel().unwrap()))
+}
+
+async fn lock_range(ctx: &mut RangeContext, ch: &IoChannel) {
+    let mut nexus = Bdev::open_by_name(NEXUS_NAME, true).unwrap();
+    let _ = nexus.lock_lba_range(ctx, ch).await;
+}
+
+async fn unlock_range(ctx: &mut RangeContext, ch: &IoChannel) {
+    let mut nexus = Bdev::open_by_name(NEXUS_NAME, true).unwrap();
+    let _ = nexus.unlock_lba_range(ctx, ch).await;
+}
+
+#[test]
+// Test acquiring and releasing a lock.
+fn lock_unlock() {
+    test_ini();
+    Reactor::block_on(async {
+        let mut nexus = Bdev::open_by_name(NEXUS_NAME, true).unwrap();
+        let mut ctx = RangeContext::new(1, 5);
+        let ch = nexus.get_channel().unwrap();
+        let _ = nexus.lock_lba_range(&mut ctx, &ch).await;
+        let _ = nexus.unlock_lba_range(&mut ctx, &ch).await;
+    });
+    test_fini();
+}
+
+#[test]
+// Test that an error is received when the lock/unlock contexts don't match
+fn lock_unlock_different_context() {
+    test_ini();
+    Reactor::block_on(async {
+        let mut nexus = Bdev::open_by_name(NEXUS_NAME, true).unwrap();
+
+        let mut ctx = RangeContext::new(1, 5);
+        let ch = nexus.get_channel().unwrap();
+        let _ = nexus.lock_lba_range(&mut ctx, &ch).await;
+
+        let mut ctx1 = RangeContext::new(1, 5);
+        let ch1 = nexus.get_channel().unwrap();
+        if nexus.unlock_lba_range(&mut ctx1, &ch1).await.is_ok() {
+            panic!("Shouldn't be able to unlock with a different context");
+        }
+    });
+    test_fini();
+}
+
+#[test]
+// Test taking out multiple locks on an overlapping block range.
+// The second lock should only succeeded after the first lock is released.
+fn multiple_locks() {
+    test_ini();
+    {
+        let reactor = Reactors::current();
+
+        // First lock
+        let (s, r) = unbounded::<()>();
+        let ctx = get_shareable_ctx(1, 10);
+        let ctx_clone = Arc::clone(&ctx);
+        let ch = get_shareable_nexus_channel();
+        let ch_clone = Arc::clone(&ch);
+        reactor.send_future(async move {
+            lock_range(
+                &mut ctx_clone.lock().unwrap(),
+                &ch_clone.lock().unwrap(),
+            )
+            .await;
+            let _ = s.send(());
+        });
+        reactor_poll!(r);
+
+        // Second lock
+        let (lock_sender, lock_receiver) = unbounded::<()>();
+        let ctx1 = get_shareable_ctx(1, 5);
+        let ctx1_clone = Arc::clone(&ctx1);
+        let ch1 = get_shareable_nexus_channel();
+        let ch_clone1 = Arc::clone(&ch1);
+        reactor.send_future(async move {
+            lock_range(
+                &mut ctx1_clone.lock().unwrap(),
+                &ch_clone1.lock().unwrap(),
+            )
+            .await;
+            trace!("Second lock succeeded");
+            let _ = lock_sender.send(());
+        });
+        reactor_poll!(100);
+
+        // First lock is held, second lock shouldn't succeed
```

Comment:
Thank you for the comments -- really helpful to understand what you are trying to test :+1:

Reply:
:+1: